src/transformers/integrations/integration_utils.py

```diff
@@ -296,28 +307,12 @@ def _objective(trial, local_trainer, checkpoint_dir=None):
         from ray.tune import CLIReporter
 
         kwargs["progress_reporter"] = CLIReporter(metric_columns=["objective"])
-    if "keep_checkpoints_num" in kwargs and kwargs["keep_checkpoints_num"] > 0:
-        # `keep_checkpoints_num=0` would disabled checkpointing
-        trainer.use_tune_checkpoints = True
-        if kwargs["keep_checkpoints_num"] > 1:
-            logger.warning(
-                f"Currently keeping {kwargs['keep_checkpoints_num']} checkpoints for each trial. "
-                "Checkpoints are usually huge, "
-                "consider setting `keep_checkpoints_num=1`."
-            )
```

Comment:
Previously, you needed to set a # of checkpoints to keep in order to do checkpointing at all -- now this is just an optional arg